### ISSUE TITLE
Card block header hierarchy

### ISSUE THREAD

**Issue by phette23** [cca/libraries_wagtail#167]
Card blocks use an h4 but appear in the main body content where I think they need to be h3. We probably cannot actually fix this without really complicated code that dynamically chooses the heading level based on other content (because whether or not there is a subheading before the card changes what level it should be) but we can at least make it correct in the most common case of h1 page title > h2 subheading > h3 card header.

This happens in [the card.html template](https://github.com/cca/libraries_wagtail/blob/main/libraries/categories/templates/categories/blocks/card.html#L4).

Highlighted by the Wagtail a11y checker. `.card:nth-child(1) > .card__header`
